src/vs/editor/browser/widget/diffEditorWidget.ts

```diff
@@ -1799,6 +1796,68 @@ const DECORATIONS = {
 
 };
 
+// {{SQL CARBON EDIT}} - helper function for reversing colors for schema compare
+/**
+	 * Reverses the decorations of the given array of decorations, so that deletes and inserts are swapped
+	 * @param decorations
+	 */
+function reverseDecorations(decorations: IModelDeltaDecoration[]): void {
```

Comment:
I moved these helper functions into diffEditorWidget.ts because there was a cyclic dependency between sql/editor/browser/diffEditorHelper and vs/editor/browser/widget/diffEditorWidget because of DECORATIONS being needed for `reverseDecorations`. Let me know if having it in a different part of this file if will make it easier for vscode merges.